### PR TITLE
Don't overwrite external service config changes when sync is running

### PR DESCRIPTION
**Pull request by eseliger**
This fixes an issue [reported by Marc on slack](https://sourcegraph.slack.com/archives/C022SPMNR0W/p1688441380321489) where changes to the external service config disappeared after a few minutes.

When this happens:
- A sync for the external service in question is in progress
- A new config is saved while the sync is still in progress

Problem:
- SyncExternalService loads the external service on entry
- Now it's syncing for an arbitrary amount of time
- And at the end, we store LastSyncAt and NextSync at values on the external service using Upsert, using the external service loaded at entry into this function

This is writing all fields from the external service passed to Upsert, which is still holding data from minutes ago, if the sync takes a while. This will effectively revert all values that changed in the meantime.

This is fixed by explicitly only writing these two fields in the syncer.

## Test plan

Wrote tests, and made sure I couldn't replicate the above behavior anymore.

### CODE REVIEW

File: internal/repos/syncer.go [hunk 1]
```diff
@@ -528,7 +528,13 @@ func (s *Syncer) SyncExternalService(
 		svc.NextSyncAt = now.Add(interval)
 		svc.LastSyncAt = now
 
-		if err := s.Store.ExternalServiceStore().Upsert(ctx, svc); err != nil {
+		// We call Update here instead of Upsert, because upsert stores all fields of the external
+		// service, and syncs take a while so changes to the external service made while this sync
+		// was running would be overwritten again.
+		if err := s.Store.ExternalServiceStore().Update(ctx, nil, svc.ID, &database.ExternalServiceUpdate{
+			LastSyncAt: &svc.LastSyncAt,
```
[the remark this answers is not in the document]
Comment (eseliger) [reply internal/repos/syncer.go:535]: that's true, I can make these individual properties

File: internal/database/external_services.go [hunk 2]
```diff
@@ -870,7 +872,7 @@ func (e *externalServiceStore) Update(ctx context.Context, ps []schema.AuthProvi
 		unrestricted := !envvar.SourcegraphDotComMode() && !gjson.GetBytes(normalized, "authorization").Exists()
 		updates = append(updates,
 			sqlf.Sprintf(
-				"config = %s, encryption_key_id = %s, next_sync_at = NOW(), unrestricted = %s, has_webhooks = %s",
```
Comment (mrnugget) [inline internal/database/external_services.go:873]: Just to double-check: you're sure that this doesn't break stuff, right?

**Review by mrnugget — APPROVED** [review]

File: internal/database/external_services.go (hunk 2 above)
Comment (eseliger) [reply internal/database/external_services.go:873]: It is readded below in the else branch of the updates.NextSyncAt. 
Do you see that changing the behavior? It seems equal to me, and a simple smoke test worked.

Comment (mrnugget) [reply internal/database/external_services.go:873]: Ah! I missed that `else` there. Yep makes sense

**Pull request merged**